### ISSUE TITLE
[a5e2d9d2] Cannot convert to AsyncDataContext at 'popup' DataContextWrapper(CaretSpecificDataContext(com.maddyhome.idea.vim.helper.EditorDataContext)). Please use CustomizedDataContext or its inheritors like SimpleDataContext

### ISSUE THREAD

**Issue by fwdekkerbot** [FWDekkerBot/intellij-randomness-issues#5]
_This issue report was generated automatically for an anonymous user. If you are this anonymous user and want this issue deleted for whatever reason, leave a comment below or <a href="https://fwdekker.com/about/">contact me</a>._

---

**User-supplied comments**
_No comments supplied._

**Stacktrace 1**
<details>
  <summary>Click to show</summary>

  ```java
  java.lang.Throwable: Cannot convert to AsyncDataContext at 'popup' DataContextWrapper(CaretSpecificDataContext(com.maddyhome.idea.vim.helper.EditorDataContext)). Please use CustomizedDataContext or its inheritors like SimpleDataContext
  	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:376)
  	at com.intellij.openapi.actionSystem.impl.Utils.checkAsyncDataContext(Utils.kt:202)
  	at com.intellij.openapi.actionSystem.impl.Utils$expandActionGroupImpl$1.invokeSuspend(Utils.kt:333)
  	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
  	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
  	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:280)
  	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
  	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
  	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
  	at com.intellij.openapi.actionSystem.impl.UtilsKt$runBlockingForActionExpand$1.invoke(Utils.kt:1200)
  	at com.intellij.openapi.actionSystem.impl.UtilsKt$runBlockingForActionExpand$1.invoke(Utils.kt:1197)
  	at com.intellij.openapi.progress.ContextKt.prepareThreadContext(context.kt:83)
  	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroupImpl(Utils.kt:1242)
  	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.kt:308)
  	at com.intellij.ui.popup.ActionStepBuilder.appendActionsFromGroup(ActionStepBuilder.java:99)
  	at com.intellij.ui.popup.ActionStepBuilder.buildGroup(ActionStepBuilder.java:65)
  	at com.intellij.ui.popup.ActionPopupStep.createActionItems(ActionPopupStep.java:126)
  	at com.intellij.ui.popup.PopupFactoryImpl$ActionGroupPopup.createStep(PopupFactoryImpl.java:281)
  	at com.intellij.ui.popup.PopupFactoryImpl$ActionGroupPopup.<init>(PopupFactoryImpl.java:239)
  	at com.intellij.ui.popup.PopupFactoryImpl$ActionGroupPopup.<init>(PopupFactoryImpl.java:221)
  	at com.intellij.ui.popup.PopupFactoryImpl$ActionGroupPopup.<init>(PopupFactoryImpl.java:188)
  	at com.intellij.ui.popup.PopupFactoryImpl.createActionGroupPopup(PopupFactoryImpl.java:373)
  	at com.jetbrains.rdserver.ui.popup.BackendBePopupFactory.createActionGroupPopup(BackendBePopupFactory.kt:111)
  	at com.intellij.openapi.ui.popup.JBPopupFactory.createActionGroupPopup(JBPopupFactory.java:259)
  	at com.jetbrains.rdserver.ui.popup.BackendBePopupFactory.createActionGroupPopup(BackendBePopupFactory.kt:165)
  	at com.fwdekker.randomness.PopupAction.actionPerformed(PopupAction.kt:55)
  	at com.maddyhome.idea.vim.helper.IjActionExecutor.executeAction$lambda$0(IjActionExecutor.kt:118)
  	at com.maddyhome.idea.vim.helper.IjActionExecutor.performDumbAwareWithCallbacks(IjActionExecutor.kt:138)
  	at com.maddyhome.idea.vim.helper.IjActionExecutor.executeAction(IjActionExecutor.kt:118)
  	at com.maddyhome.idea.vim.vimscript.model.commands.ActionCommand.executeAction(ActionCommand.kt:46)
  	at com.maddyhome.idea.vim.vimscript.model.commands.ActionCommand.processCommand$lambda$0(ActionCommand.kt:40)
  	at com.intellij.openapi.wm.impl.FocusManagerImpl.lambda$doWhenFocusSettlesDown$4(FocusManagerImpl.java:174)
  	at com.intellij.util.ui.EdtInvocationManager.invokeLaterIfNeeded(EdtInvocationManager.java:33)
  	at com.intellij.ide.IdeEventQueue.ifFocusEventsInTheQueue(IdeEventQueue.kt:226)
  	at com.intellij.ide.IdeEventQueue.executeWhenAllFocusEventsLeftTheQueue(IdeEventQueue.kt:192)
  	at com.intellij.openapi.wm.impl.FocusManagerImpl.doWhenFocusSettlesDown(FocusManagerImpl.java:170)
  	at com.maddyhome.idea.vim.helper.UiHelper.runAfterGotFocus(UiHelper.kt:32)
  	at com.maddyhome.idea.vim.newapi.IjVimApplication.runAfterGotFocus(IjVimApplication.kt:82)
  	at com.maddyhome.idea.vim.vimscript.model.commands.ActionCommand.processCommand(ActionCommand.kt:40)
  	at com.maddyhome.idea.vim.vimscript.model.commands.Command.runCommand(Command.kt:99)
  	at com.maddyhome.idea.vim.vimscript.model.commands.Command.access$runCommand(Command.kt:34)
  	at com.maddyhome.idea.vim.vimscript.model.commands.Command$execute$runCommand$1.invoke(Command.kt:78)
  	at com.maddyhome.idea.vim.vimscript.model.commands.Command$execute$runCommand$1.invoke(Command.kt:78)
  	at com.maddyhome.idea.vim.newapi.IjVimApplication.runReadAction$lambda$4(IjVimApplication.kt:74)
  	at com.intellij.openapi.application.impl.RwLockHolder.runReadAction(RwLockHolder.kt:271)
  	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:845)
  	at com.maddyhome.idea.vim.newapi.IjVimApplication.runReadAction(IjVimApplication.kt:74)
  	at com.maddyhome.idea.vim.vimscript.model.commands.Command.execute(Command.kt:81)
  	at com.maddyhome.idea.vim.vimscript.Executor.execute(Executor.kt:53)
  	at com.maddyhome.idea.vim.group.ProcessGroup.processExEntry(ProcessGroup.kt:135)
  	at com.maddyhome.idea.vim.action.ex.ProcessExEntryAction.execute(ProcessExEntryAction.kt:33)
  	at com.maddyhome.idea.vim.handler.VimActionHandler.baseExecute(VimActionHandler.kt:87)
  	at com.maddyhome.idea.vim.handler.EditorActionHandlerBase.doExecute(EditorActionHandlerBase.kt:129)
  	at com.maddyhome.idea.vim.handler.EditorActionHandlerBase.access$doExecute(EditorActionHandlerBase.kt:43)
  	at com.maddyhome.idea.vim.handler.EditorActionHandlerBase$execute$action$1.invoke(EditorActionHandlerBase.kt:88)
  	at com.maddyhome.idea.vim.handler.EditorActionHandlerBase$execute$action$1.invoke(EditorActionHandlerBase.kt:88)
  	at com.maddyhome.idea.vim.handler.EditorActionHandlerBase.execute(EditorActionHandlerBase.kt:106)
  	at com.maddyhome.idea.vim.helper.IjActionExecutor.executeVimAction$lambda$2(IjActionExecutor.kt:226)
  	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:216)
  	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:177)
  	at com.maddyhome.idea.vim.helper.IjActionExecutor.executeVimAction(IjActionExecutor.kt:224)
  	at com.maddyhome.idea.vim.KeyHandler$ActionRunner.run(KeyHandler.kt:317)
  	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:225)
  	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:177)
  	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:167)
  	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:153)
  	at com.maddyhome.idea.vim.helper.IjActionExecutor.executeCommand(IjActionExecutor.kt:210)
  	at com.maddyhome.idea.vim.KeyHandler.executeCommand(KeyHandler.kt:241)
  	at com.maddyhome.idea.vim.KeyHandler.finishedCommandPreparation$vim_engine(KeyHandler.kt:167)
  	at com.maddyhome.idea.vim.KeyHandler$processKey$1$3.invoke(KeyHandler.kt:138)
  	at com.maddyhome.idea.vim.KeyHandler$processKey$1$3.invoke(KeyHandler.kt:137)
  	at com.maddyhome.idea.vim.KeyProcessResult$SynchronousKeyProcessBuilder$build$1.invoke(KeyHandler.kt:451)
  	at com.maddyhome.idea.vim.KeyProcessResult$SynchronousKeyProcessBuilder$build$1.invoke(KeyHandler.kt:448)
  	at com.maddyhome.idea.vim.KeyProcessResult$Executable.execute(KeyHandler.kt:411)
  	at com.maddyhome.idea.vim.KeyHandler.handleKey(KeyHandler.kt:91)
  	at com.maddyhome.idea.vim.key.ToKeysMappingInfo.execute(MappingInfo.kt:100)
  	at com.maddyhome.idea.vim.command.MappingProcessor.processCompleteMappingSequence(MappingProcessor.kt:209)
  	at com.maddyhome.idea.vim.command.MappingProcessor.access$processCompleteMappingSequence(MappingProcessor.kt:28)
  	at com.maddyhome.idea.vim.command.MappingProcessor$handleCompleteMappingSequence$2.invoke(MappingProcessor.kt:191)
  	at com.maddyhome.idea.vim.command.MappingProcessor$handleCompleteMappingSequence$2.invoke(MappingProcessor.kt:191)
  	at com.maddyhome.idea.vim.KeyProcessResult$SynchronousKeyProcessBuilder$build$1.invoke(KeyHandler.kt:451)
  	at com.maddyhome.idea.vim.KeyProcessResult$SynchronousKeyProcessBuilder$build$1.invoke(KeyHandler.kt:448)
  	at com.maddyhome.idea.vim.KeyProcessResult$Executable.execute(KeyHandler.kt:411)
  	at com.maddyhome.idea.vim.KeyHandler.handleKey(KeyHandler.kt:91)
  	at com.maddyhome.idea.vim.KeyHandler.handleKey(KeyHandler.kt:72)
  	at com.maddyhome.idea.vim.action.VimShortcutKeyAction.actionPerformed(VimShortcutKeyAction.kt:82)
  	at com.intellij.openapi.actionSystem.AnActionWrapper.actionPerformed(AnActionWrapper.java:40)
  	at com.intellij.openapi.actionSystem.ex.ActionUtil.doPerformActionOrShowPopup(ActionUtil.kt:304)
  	at com.intellij.openapi.keymap.impl.ActionProcessor.performAction(ActionProcessor.java:32)
  	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher$actionProcessor$1.performAction(IdeKeyEventDispatcher.kt:495)
  	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcherKt.doPerformActionInner$lambda$4$lambda$3(IdeKeyEventDispatcher.kt:844)
  	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:106)
  	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:95)
  	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcherKt.doPerformActionInner$lambda$4(IdeKeyEventDispatcher.kt:844)
  	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.performWithActionCallbacks(ActionManagerImpl.kt:1161)
  	at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareWithCallbacks(ActionUtil.kt:321)
  	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcherKt.doPerformActionInner(IdeKeyEventDispatcher.kt:842)
  	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcherKt.access$doPerformActionInner(IdeKeyEventDispatcher.kt:1)
  	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processAction$intellij_platform_ide_impl(IdeKeyEventDispatcher.kt:571)
  	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processAction(IdeKeyEventDispatcher.kt:508)
  	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processActionOrWaitSecondStroke(IdeKeyEventDispatcher.kt:447)
  	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.inInitState(IdeKeyEventDispatcher.kt:440)
  	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.dispatchKeyEvent(IdeKeyEventDispatcher.kt:302)
  	at com.intellij.ide.IdeEventQueue.dispatchKeyEvent(IdeEventQueue.kt:623)
  	at com.intellij.ide.IdeEventQueue._dispatchEvent$lambda$11(IdeEventQueue.kt:585)
  	at com.intellij.openapi.application.impl.RwLockHolder.runWithEnabledImplicitRead(RwLockHolder.kt:138)
  	at com.intellij.openapi.application.impl.RwLockHolder.runWithImplicitRead(RwLockHolder.kt:129)
  	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:585)
  	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:77)
  	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:362)
  	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:361)
  	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:843)
  	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:361)
  	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:356)
  	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:1022)
  	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:114)
  	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:1022)
  	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$7(IdeEventQueue.kt:356)
  	at com.intellij.openapi.application.impl.RwLockHolder.runIntendedWriteActionOnCurrentThread(RwLockHolder.kt:209)
  	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:830)
  	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:398)
  	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
  	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
  	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
  	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
  	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
  	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
  
  ```

</details>

**Version information**
- Randomness version: 3.3.0
- IDE version: WS-241.15989.105
- Operating system: Mac OS X
- Java version: 17.0.10

**Comment by fwdekker** [comment]
This looks like an error with the interaction between the Vim plugin and Randomness. I'll install that and see if I can do something to replicate this.

**Comment by fwdekker** [comment]
As far as I can tell, this is an issue with the IdeaVim plugin. I have reproduced the error and described it in detail in [a comment on the existing issue VIM-3364](https://youtrack.jetbrains.com/issue/VIM-3364/Exception-with-mapped-Generate-action#focus=Comments-27-9785570.0-0).

**Comment by fwdekker** [comment]
I [received message](https://youtrack.jetbrains.com/issue/VIM-3364/Exception-with-mapped-Generate-action#focus=Comments-27-10228528.0-0) that this issue should have been resolved as of IdeaVim 2.16.0-eap.6. You can either [update IdeaVim to use an EAP release](https://github.com/JetBrains/ideavim#get-early-access), or wait for the fix to land in the next stable release of IdeaVim.

Also, I am unable to reproduce the bug with IdeaVim 2.15.2 in IntelliJ Community 2024.1.2. Maybe the fix was actually already done earlier? Either way, I think this bug is now completely solved :-)
